evap/results/views.py

```diff
@@ -100,18 +100,12 @@ def course_detail(request, semester_id, course_id):
     add_warnings(course, course_result)
 
     # split course_result into different lists
+    contributor_contribution_results = [contribution_result for contribution_result in course_result.contribution_results if contribution_result.contributor is not None]
+    general_contribution_result = next(contribution_result for contribution_result in course_result.contribution_results if contribution_result.contributor is None)
```

Comment:
is this the same as `[questionnaire_result for contribution_result in course_result.contribution_results if contribution_result.contributor is None for questionnaire_result in contribution_result.questionnaire_results]`?

Reply:
although much longer i might prefer this one because one can see that we're collecting the questionnaire results.

Reply:
please check again, i reverted the commit that made this part more confusing.

Reply:
i don't remember exactly but i think that this replaced about 10 other lines of code. so you didn't like two two-line solution or why did you revert the one commit?

Reply:
check out the last commit, this is what it would look like. it's not shorter, it's just less lines of code.

Reply:
okay, that probably doesn't really make it easier to understand. i'm fine with either option.